src/replay/consensus.zig

```diff
@@ -1636,3 +1719,307 @@ test "computeBankStats - same weight selects lower slot" {
+            try dgop.value_ptr.put(testing.allocator, slot);
+        }
+    }
+
+    var db = try TestDB.init(@src());
+    defer db.deinit();
+    var registry = sig.prometheus.Registry(.{}).init(testing.allocator);
+    defer registry.deinit();
+    var lowest_cleanup_slot = RwMux(Slot).init(0);
+    var max_root = std.atomic.Value(Slot).init(0);
+    var ledger_reader = try LedgerReader.init(
+        testing.allocator,
+        .noop,
+        db,
+        &registry,
+        &lowest_cleanup_slot,
+        &max_root,
+    );
+    var ledger_writer = try LedgerResultWriter.init(
+        testing.allocator,
+        .noop,
+        db,
+        &registry,
+        &lowest_cleanup_slot,
+        &max_root,
+    );
+
+    var slot_data: SlotData = .empty;
+    defer slot_data.deinit(testing.allocator);
+    var ancestor_hashes_replay_update_sender =
+        try sig.sync.Channel(AncestorHashesReplayUpdate).create(testing.allocator);
+    defer ancestor_hashes_replay_update_sender.destroy();
+
+    var replay_tower = try createTestReplayTower(1, 0.67);
+    defer replay_tower.deinit(testing.allocator);
+
+    const versioned_stakes = try testEpochStakes(
+        testing.allocator,
+        fixture.vote_pubkeys.items,
+        10000,
+        random,
+    );
+    var epoch_tracker: EpochTracker = .{ .epochs = .empty, .schedule = .DEFAULT };
+    try epoch_tracker.epochs.put(testing.allocator, 0, .{
+        .hashes_per_tick = null,
+        .ticks_per_slot = 1,
+        .ns_per_slot = 1,
+        .genesis_creation_time = 0,
+        .slots_per_year = 1,
+        .stakes = versioned_stakes,
+        .rent_collector = sig.core.RentCollector.DEFAULT,
+    });
+
+    var tsm1 = sig.accounts_db.ThreadSafeAccountMap.init(testing.allocator);
+    defer tsm1.deinit();
+    var slot_history1 = try sig.runtime.sysvar.SlotHistory.init(testing.allocator);
+    defer slot_history1.deinit(testing.allocator);
+    const ser1 = try sig.bincode.writeAlloc(testing.allocator, slot_history1, .{});
+    defer testing.allocator.free(ser1);
+    try tsm1.put(0, sig.runtime.sysvar.SlotHistory.ID, .{
+        .lamports = 1,
+        .data = ser1,
+        .owner = sig.core.Pubkey.ZEROES,
+        .executable = false,
+        .rent_epoch = 0,
+    });
+
+    const deps: ConsensusDependencies = .{
+        .allocator = testing.allocator,
+        .logger = .noop,
+        .replay_tower = &replay_tower,
+        .progress_map = &fixture.progress,
+        .slot_tracker = &fixture.slot_tracker,
+        .epoch_tracker = &epoch_tracker,
+        .fork_choice = &fixture.fork_choice,
+        .ledger_reader = &ledger_reader,
+        .ledger_result_writer = &ledger_writer,
+        .ancestors = &ancestors,
+        .descendants = &descendants,
+        .vote_account = fixture.vote_pubkeys.items[0],
+        .slot_history_accessor = &(blk: {
+            var accessor = SlotHistoryAccessor.init(tsm1.accountReader());
+            break :blk accessor;
+        }),
```

Comment:
I know this is a test but this is a red flag. Prefer just declaring a variable outside instead of relying on this pointer to a temporary being mutable (which it possibly shouldn't be, and is just missing a compile error due to lvalues being propagated like this being a relatively recent-ish behavior, since 0.13 I think, and not set in stone).

Reply:
[ad7506f](https://github.com/Syndica/sig/pull/895/commits/ad7506f698e4dd4f43c9d5464d9b452993790bac)